### PR TITLE
Volcano adapts to the k8s v1.30

### PR DESCRIPTION
**Pull request by guoqinwill**
Fix https://github.com/volcano-sh/volcano/issues/3523
cherry pick and update：https://github.com/volcano-sh/volcano/pull/3534/commits

### CODE REVIEW

**Comment by lowang-bh** [comment]
It is better to squash 3534's commits.

File: pkg/scheduler/capabilities/volumebinding/binder_test.go [hunk 1]
```diff
@@ -2457,12 +2457,12 @@ func TestGetEligibleNodes(t *testing.T) {
 		eligibleNodes := testEnv.binder.GetEligibleNodes(logger, scenario.pvcs)
 
 		// Validate
-		if equality.Semantic.DeepEqual(scenario.eligibleNodes, eligibleNodes) {
+		if reflect.DeepEqual(scenario.eligibleNodes, eligibleNodes) {
```
Comment (lowang-bh) [inline pkg/scheduler/capabilities/volumebinding/binder_test.go:2460]: Is this changed back to reflect.DeepEqual?

**Comment by guoqinwill** [comment]
> It is better to squash 3534's commits.

done

File: pkg/scheduler/capabilities/volumebinding/volume_binding_test.go [hunk 2]
```diff
@@ -19,6 +19,7 @@ package volumebinding
 import (
 	"context"
 	"testing"
+	"volcano.sh/volcano/cmd/scheduler/app/options"
```
[the remark this answers is not in the document]
Comment (guoqinwill) [reply pkg/scheduler/capabilities/volumebinding/volume_binding_test.go]: done

File: pkg/scheduler/capabilities/volumebinding/binder_test.go (hunk 1 above)
Comment (guoqinwill) [reply pkg/scheduler/capabilities/volumebinding/binder_test.go]: There was a deviation in conflict resolution and it has been fixed.

**Review by william-wang — APPROVED** [review]
/approve

**Pull request merged**